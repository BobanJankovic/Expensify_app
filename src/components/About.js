import React from 'react';
export const About = () => (
    <div>
    This is from  About
    </div>
);

 