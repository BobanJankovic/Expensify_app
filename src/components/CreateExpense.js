import React from 'react';
import { connect } from 'react-redux';
import AddForm from '../components/AddExpense';
import { addExpense } from '../actions/expenses';


const CreateExpense = (props) => (
    <div>
    <AddForm
      onSubmit={(expense) => {
        props.dispatch(addExpense(expense));
        props.history.push('/');
        
      }}
    />
    </div>
);

 const mapStateToProps = (state)=> {
  return {
    expenses:state.expenses,
    filters:state.filters
  };
};
export default connect()(CreateExpense);