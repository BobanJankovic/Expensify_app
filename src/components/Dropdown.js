import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import {sortByAmount} from '../actions/filters';
import {sortByDate} from '../actions/filters';

const Dropdown= (props)=> (
  <div>
    <select value={props.filters.sortBy} 
      onChange={
                (e)=>{
                  if (e.target.value==="date") {
                    props.dispatch(sortByDate())
                  } else if ((e.target.value==="amount")){
                    props.dispatch(sortByAmount())
                  }
                }
              }
      >
      <option value="amount" >Amount</option>
      <option value="date" >Date</option>
      
    </select>
  </div>
);

const mapStateToProps = (state)=> {
  return {
    expenses:selectExpenses(state.expenses,state.filters),
    filters:state.filters
  };
};

export default connect(mapStateToProps)(Dropdown);
