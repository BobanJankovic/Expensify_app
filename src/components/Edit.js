import React from 'react';
import { connect } from 'react-redux';
import AddForm from './AddExpense';

import { startEditExpense,addExpense, removeExpense, startRemoveExpense } from '../actions/expenses';


const Edit = (props) => {
   console.log(props);
  return (
    <div>
    
      
      <AddForm
      //  This is edit od {props.expense.id}{props.expense.id} zbog linije koda 12 mi se vrednosti pojave u poljima koje treba da izmenim
      //props.expense pristupio sam iz stora i poslao je preko propsa
      expenseedit={props.expense}
       onSubmit={(expensed) => {
        
         props.dispatch(startEditExpense(props.expense.id, expensed));
          //props.dispatch(addExpense(expense));
          props.history.push('/');
         
        }}
         />
          <button  onClick={()=>{
            props.dispatch(startRemoveExpense({ id: props.expense.id}))
            props.history.push('/');
            }}>remove</button>
    </div>
  );
};


const mapStateToProps = (state, props) => {
  console.log(props);
  return {
    
    expense: state.expenses.find(expense => expense.id === props.match.params.id )
  };
};

export default connect(mapStateToProps)(Edit);
   
   /*import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startEditExpense, removeExpense } from '../actions/expenses';

const EditExpensePage = (props) => {
  return (
    <div>
      <ExpenseForm
        expense={props.expense}
        onSubmit={(expense) => {
          props.dispatch(startEditExpense(props.expense.id, expense));
          props.history.push('/');
        }}
      />
      <button onClick={() => {
        props.dispatch(removeExpense({ id: props.expense.id }));
        props.history.push('/');
      }}>Remove</button>
    </div>
  );
};

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
  };
};

export default connect(mapStateToProps)(EditExpensePage); */