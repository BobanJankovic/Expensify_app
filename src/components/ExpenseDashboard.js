import React from 'react';

import ExpensePage from '../components/ExpensePage';

export const ExpenseDashboard = () => (
    <div>
 
    <ExpensePage />

    </div>
);