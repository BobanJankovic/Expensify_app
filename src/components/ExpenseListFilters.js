

import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import {setTextFilter} from '../actions/filters';
//connect redux component to the redux store
const ExpenseListFilter= (props)=> (
<div>
  <div>
  
  <input type="text" value={props.filters.text} onChange={(e)=>{
      console.log(e.target.value);
      props.dispatch(setTextFilter(e.target.value));
  }}/>
  </div>
  
  
  </div>

);

const mapStateToProps = (state)=> {
  return {
    filters:state.filters
    
  };
};
export default connect(mapStateToProps)(ExpenseListFilter);








