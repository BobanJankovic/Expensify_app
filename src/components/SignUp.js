import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';


export const SignUp = (props) => (
  <div className="loginPage">
  <h3>Login page</h3>
    <button onClick={props.startLogin}>Sign Up</button>
  </div>
);
  

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(SignUp);




