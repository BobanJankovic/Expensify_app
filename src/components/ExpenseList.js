import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from "../components/ExpenseListItem";
import ExpenseListFilter from "../components/ExpenseListFilters";
import Dropdown from "../components/Dropdown";


//connect redux component to the redux store
const ExpenseList= (props)=> (
<div>
  <h1>Boban Jankovic je car</h1>
  <Dropdown />
  <ExpenseListItem />
  <ExpenseListFilter />

</div>
);

const mapStateToProps = (state)=> {
  return {
    expenses:state.expenses,
    filters:state.filters
  };
};
export default connect(mapStateToProps)(ExpenseList);

