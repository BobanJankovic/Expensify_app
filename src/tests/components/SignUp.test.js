import React from "react";
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import {SignUp} from '../../components/SignUp';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });


test('should render SignUp correctly', () => {
const wrapper = shallow(<SignUp />);
expect(toJSON(wrapper)).toMatchSnapshot();
expect(wrapper.find('h3').text()).toBe('Login page');
});

test('should call startLogin on button click', () => {
  const onSubmitSpy = jest.fn();
  const wrapper = shallow(<SignUp startLogin={ onSubmitSpy }  />);
  
 wrapper.find('button').simulate('click');
  expect(onSubmitSpy).toHaveBeenCalled();
});