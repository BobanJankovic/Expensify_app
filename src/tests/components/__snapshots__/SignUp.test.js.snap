// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SignUp correctly 1`] = `
<div
  className="loginPage"
>
  <h3>
    Login page
  </h3>
  <button>
    Sign Up
  </button>
</div>
`;
